pkg/proxy/iptables/proxier_test.go

```diff
@@ -1590,30 +1925,53 @@ func makeServiceMap(proxier *Proxier, allServices ...*api.Service) {
 	proxier.servicesSynced = true
 }
 
-func compareEndpointsMaps(t *testing.T, tci int, newMap, expected map[proxy.ServicePortName][]*endpointsInfo) {
+func compareEndpointsMaps(t *testing.T, testCase string, newMap, expected map[proxy.ServicePortName][]*endpointsInfo) {
 	if len(newMap) != len(expected) {
-		t.Errorf("[%d] expected %d results, got %d: %v", tci, len(expected), len(newMap), newMap)
+		t.Errorf("[%s] expected %d results, got %d: %v", testCase, len(expected), len(newMap), newMap)
 	}
 	for x := range expected {
 		if len(newMap[x]) != len(expected[x]) {
-			t.Errorf("[%d] expected %d endpoints for %v, got %d", tci, len(expected[x]), x, len(newMap[x]))
+			t.Errorf("[%s] expected %d endpoints for %v, got %d", testCase, len(expected[x]), x, len(newMap[x]))
 		} else {
 			for i := range expected[x] {
 				if *(newMap[x][i]) != *(expected[x][i]) {
-					t.Errorf("[%d] expected new[%v][%d] to be %v, got %v", tci, x, i, expected[x][i], newMap[x][i])
+					t.Errorf("[%s] expected new[%v][%d] to be %v, got %v", testCase, x, i, expected[x][i], newMap[x][i])
 				}
 			}
 		}
 	}
 }
 
 func Test_updateEndpointsMap(t *testing.T) {
+	prefixes := [][]string{{
+		"1.1.1.",
+		"2.2.2.",
+		"3.3.3.",
+		"4.4.4.",
+	}, {
+		"2001:db8:1::",
+		"2001:db8:2::",
+		"2001:db8:3::",
+		"2001:db8:4::",
+	}}
+	for _, prefs := range prefixes {
+		updateEndpointsMapTest(t, prefs)
+	}
+}
+
+func updateEndpointsMapTest(t *testing.T, prefixes []string) {
 	var nodeName = testHostname
 
+	testIP := func(prefixIndex int, hostNum int) string {
+		// Assumes prefixIndex is a 1-based index for readability
+		// (so that its value will match corresponding prefix strings)
+		return prefixes[prefixIndex-1] + strconv.Itoa(hostNum)
+	}
+
 	unnamedPort := func(ept *api.Endpoints) {
 		ept.Subsets = []api.EndpointSubset{{
 			Addresses: []api.EndpointAddress{{
-				IP: "1.1.1.1",
+				IP: testIP(1, 1),
```

Comment:
I really think the indirection of the testIP function makes the tests harder to read.